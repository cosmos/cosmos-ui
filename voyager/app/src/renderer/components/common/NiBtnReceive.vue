<template lang="pug">
btn.btn-receive(
  @click.native.stop="openModal"
  value="Receive"
  icon="file_download"
  icon-pos="right"
  color="primary")
</template>

<script>
import { clipboard } from "electron"
import Btn from "@nylira/vue-button"
export default {
  components: { Btn },
  methods: {
    openModal() {
      this.$store.commit("setModalReceive", true)
    }
  }
}
</script>
