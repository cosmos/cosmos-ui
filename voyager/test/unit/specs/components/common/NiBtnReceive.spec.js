import setup from "../../../helpers/vuex-setup"
import NiBtnReceive from "renderer/components/common/NiBtnReceive"

describe("NiBtnReceive", () => {
  let wrapper, store
  let { mount } = setup()

  beforeEach(() => {
    let instance = mount(NiBtnReceive)
    wrapper = instance.wrapper
    store = instance.store
  })

  it("has the expected html structure", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should commit the receive modal", () => {
    wrapper.vm.openModal()
    expect(store.commit).toHaveBeenCalledWith("setModalReceive", true)
  })
})
