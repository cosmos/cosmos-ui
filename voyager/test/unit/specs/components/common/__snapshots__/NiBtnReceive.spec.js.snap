// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NiBtnReceive has the expected html structure 1`] = `"<button class=\\"ni-btn btn-receive\\"><span class=\\"ni-btn__container ni-btn__icon-right ni-btn--primary\\"><i aria-hidden=\\"true\\" class=\\"ni-btn__icon material-icons\\">file_download</i><!----><span class=\\"ni-btn__value\\">Receive</span></span></button>"`;
